packages/hdwallet-portis/src/portis.ts

```diff
@@ -82,11 +203,90 @@ export function isPortis(wallet: HDWallet): wallet is PortisHDWallet {
   return typeof wallet === 'object' && (wallet as any)._isPortis === true
 }
 
-export class PortisHDWallet implements HDWallet, ETHWallet {
+export class PortisHDWallet implements HDWallet, ETHWallet, BTCWallet {
+
+  // btc stuff
+
+
+  public async btcGetAddress (msg: BTCGetAddress): Promise<string> {
+    
+    const change = msg.addressNList[3]
+    const index = msg.addressNList[4]
+
+    const b32string = addressNListToBIP32(msg.addressNList)
+    const hardPath = b32string.slice(0, b32string.lastIndexOf(`'`)+1)
```

Comment:
Don't do string math. Use `hardenedPath` to get the hardened part of the addressNList.

Also, don't assume that the relpath will always include 2 entries.